### PR TITLE
Persist V13

### PR DESCRIPTION
**Pull request by mbarbet**
TODO :

- [x]  Find a solution to list data by zone with firestore implementation
- [x]  Fix delete endpoint for hibernate
- [x]  Add some test to deal with Groups/owner notion, valid with hibernate and firestore
- [x]  Add /groups endpoint to list group from zone
- [x]  For Update endpoint add check on lastDateUpdate

### CODE REVIEW

File: arlas-persistence-core/src/main/java/io/arlas/persistence/server/core/PersistenceService.java [hunk 1]
```diff
@@ -19,21 +19,64 @@
...
+    }
+
+     static boolean isReaderOnData(IdentityParam identityParam, Data data) {
+        List<String> readers =  new ArrayList<>(data.getDocReaders());
+        return data.getDocOrganization().equals(identityParam.organization) &&
+                (data.getDocOwner().equals(identityParam.userId) ||
+                        intersect(identityParam.groups, readers));
+    }
+
+     static boolean isWritersOnData(IdentityParam identityParam, Data data) {
```
Comment (alainbodiguel) [inline arlas-persistence-core/src/main/java/io/arlas/persistence/server/core/PersistenceService.java:75]: isWriterOnData ? (without 's' on Writers)

File: conf/configuration.yaml [hunk 2]
```diff
@@ -72,17 +72,20 @@ arlas_auth:
   public_uris: [${ARLAS_AUTH_PUBLIC_URIS:-swagger,swagger.json}]
   certificate_file: ${ARLAS_AUTH_CERT_FILE:-/opt/app/arlas.pem}
 
-key_header: ${ARLAS_PERSISTENCE_KEY_HEADER:-X-Forwarded-User}
+
 persistence_engine: ${ARLAS_PERSISTENCE_ENGINE:-hibernate}
-firestore_collection: ${ARLAS_PERSISTENCE_FIRESTORE_COLLECTION:-}
+firestore_collection: ${ARLAS_PERSISTENCE_FIRESTORE_COLLECTION:-test}
+arlas_groups_header:  ${ARLAS_GROUPS_HEADER:-arlas_groups}
```
Comment (alainbodiguel) [inline conf/configuration.yaml:78]: We might set the HEADER env and forget to change the ARLAS_PERSISTENCE_CORS_ALLOWED_HEADERS...

File: docker/docker-files/pgCreateTable.sql [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
 );
-CREATE INDEX doc_key_idx ON user_data (docKey, docType);
\ No newline at end of file
+CREATE UNIQUE INDEX doc_key_idx_orga ON user_data (docKey, docZone,docOrganization);
+
+DROP TABLE IF EXISTS user_data_readers;
+
+CREATE TABLE user_data_readers (
+    data_id VARCHAR(255) ,
+    readers VARCHAR(255) NOT NULL,
```
Comment (alainbodiguel) [inline docker/docker-files/pgCreateTable.sql:18]: is it one reader or several readers?

File: docker/docker-files/pgCreateTable.sql [hunk 4]
```diff
@@ -1,10 +1,30 @@
...
+    FOREIGN KEY (data_id) REFERENCES user_data (id)
+);
+
+DROP TABLE IF EXISTS user_data_writers;
+
+CREATE TABLE user_data_writers (
+    data_id VARCHAR(255) ,
+    writers VARCHAR(255) NOT NULL,
```
Comment (alainbodiguel) [inline docker/docker-files/pgCreateTable.sql:26]: is it one writer or several writers?

File: conf/configuration.yaml (hunk 2 above)
Comment (mbarbet) [reply conf/configuration.yaml:78]: A check is apply on start and an exception is throwed if arlas header are missing in allowed headers in class ArlasPersistenceServerConfiguration.java

File: docker/docker-files/pgCreateTable.sql (hunk 4 above)
Comment (mbarbet) [reply docker/docker-files/pgCreateTable.sql:26]: This table contains one line by couple data_id / group allowed to write; so i remove the 's' in the name of the table and in the name of the column ?

File: docker/docker-files/pgCreateTable.sql (hunk 3 above)
Comment (mbarbet) [reply docker/docker-files/pgCreateTable.sql:18]: This table contains one line by couple data_id / group allowed to read; so i remove the 's' in the name of the table and in the name of the column ?

Comment (alainbodiguel) [reply docker/docker-files/pgCreateTable.sql:18]: I would remove the 's' from the column name

File: docker/docker-files/pgCreateTable.sql (hunk 4 above)
Comment (alainbodiguel) [reply docker/docker-files/pgCreateTable.sql:26]: I would remove the 's' from the column name

**Pull request merged**